### PR TITLE
TAS-1094/Implemented referral system API

### PR DESCRIPTION
**Pull request by Odeyaloleg**

### CODE REVIEW

File: src/models/referring/index.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import sql from 'sql-template-tag'
+import { app } from '../../index.js'
+import { EntryError } from '../../utils/errors.js'
+
+const insert = async (referred_profile_id, referred_by_id) => {
```
Comment (jeyem) [inline src/models/referring/index.js:5]: code style use camel case  on variable and function parameter use snake case on objects

File: src/models/referring/index.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        RETURNING *
+      `)
+    return rows[0]
+  } catch (err) {
+    throw new EntryError(err.message)
+  }
+}
+
+const get = async (referred_profile_id) => {
+  return app.db?.get(sql`SELECT referred_by_id FROM referrings WHERE referred_profile_id=${referred_profile_id} AND referred_at < now() - INTERVAL '1 year'`)
```
Comment (jeyem) [inline src/models/referring/index.js:19]: break query in 3 line please, and change parameter naming, Thanks

File: src/routes/organization.js [hunk 3]
```diff
@@ -45,6 +46,9 @@ router.post('/', loginRequired, async (ctx) => {
   }
   ctx.body = await Org.insert(ctx.user.id, ctx.request.body)
   if (ctx.query.auto_member !== 'false') await Org.addMember(ctx.body.id, ctx.user.id)
+
+  const referred_by_id = ctx.request.body?.referredBy
```
Comment (jeyem) [inline src/routes/organization.js:50]: camel case variable

**Review by jeyem — CHANGES_REQUESTED** [review]

File: migrations/sqls/20240228103534-add-user-identity-verified-up.sql [hunk 4]
```diff
@@ -0,0 +1,7 @@
+ALTER TABLE users ADD COLUMN identity_verified boolean DEFAULT false;
+
+CREATE TABLE IF NOT EXISTS referrings (
+  referred_identity_id uuid PRIMARY KEY NOT NULL REFERENCES identities(id),
+  referred_by_id uuid NOT NULL REFERENCES users(id),
```
Comment (jeyem) [inline migrations/sqls/20240228103534-add-user-identity-verified-up.sql:5]: I refer this to identity too , to make it more general cause referring by org would be possible in future some this put BE open up to do that too if it's been needed

File: src/routes/organization.js [hunk 5]
```diff
@@ -43,8 +45,18 @@ router.post('/', loginRequired, async (ctx) => {
   if (!config.mail.allowTest && isTestEmail(ctx.request.body.email)) {
     throw new ValidationError('Invalid email')
   }
+
+  const referredById = ctx.request.body?.referredBy
+  if (referredById) {
+    const user = await User.get(referredById)
+
+    if (!user.identity_verified)
```
Comment (Odeyaloleg) [inline src/routes/organization.js:53]: @jeyem Is this fine or better create a function `User.getVerified` to move this check to database layer?

File: migrations/sqls/20240228103534-add-user-identity-verified-up.sql [hunk 6]
```diff
@@ -0,0 +1,11 @@
+ALTER TABLE users ADD COLUMN identity_verified boolean DEFAULT false;
+
+CREATE TABLE IF NOT EXISTS referrings (
+  referred_identity_id uuid PRIMARY KEY NOT NULL REFERENCES identities(id),
+  referred_by_id uuid NOT NULL REFERENCES identities(id),
+  created_at timestamptz DEFAULT now() NOT NULL
+);
+
+ALTER TABLE offers
```
Comment (Odeyaloleg) [inline migrations/sqls/20240228103534-add-user-identity-verified-up.sql:9]: Not sure about correctness of this decision

**Review by jeyem — CHANGES_REQUESTED** [review]

**Pull request merged**